File: app/src/main/java/com/example/myapplication/rawresponses/MultipleTournamentsJSON.kt
package com.example.myapplication.rawresponses

data class MultipleTournamentsJSON(
    val _embedded: TournamentEntitiesJSON,
    val _links: LinksJSON,
    val page: PageJSON
)

data class TournamentEntitiesJSON(
    val tournamentEntities: List<TournamentEntityJSON>
)

data class TournamentEntityJSON(
    val id: Long,
    val playersNumber: Int,
    val tournamentDescription: String,
    val tournamentMode: String,
    val adminId: Long,
    val _links: LinksJSON
)

data class LinksJSON(
    val self: HrefJSON,
    val tournamentEntity: HrefJSON? = null,
    val profile: HrefJSON? = null
)

File: web-client/src/app/data/rawresponses/multiple/multiple-tournaments-json.ts
import {LinksJSON} from '../links-json';
import {PageJSON} from '../page-json';
import {TournamentJSON} from '../single/tournament-json';

export interface MultipleTournamentsJSON {
  readonly _embedded: TournamentEmbeddedJSON;
  readonly _links: LinksJSON;
  readonly page: PageJSON;
}


export interface TournamentEmbeddedJSON {
  readonly tournamentEntities: TournamentJSON[];
}
